### PR TITLE
fix sqlException.getCause isNull

### LINKED ISSUE #6583: sqlException.getCause is null
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### PR DESCRIPTION
**Pull request by cmonkey**
Fixes #6583.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static MySQLErrPacket newInstance(final int sequenceId, final Exception c
         if (cause instanceof SQLException) {
             SQLException sqlException = (SQLException) cause;
             return null != sqlException.getSQLState() ? new MySQLErrPacket(sequenceId, sqlException.getErrorCode(), sqlException.getSQLState(), sqlException.getMessage())
-                : new MySQLErrPacket(sequenceId, MySQLServerErrorCode.ER_INTERNAL_ERROR, sqlException.getCause().getMessage());
+                : new MySQLErrPacket(sequenceId, MySQLServerErrorCode.ER_INTERNAL_ERROR, cause.getMessage());
```
Comment (tristaZero) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: When `null == sqlException.getSQLState() `, `null == sqlException.getCause()` but `cause.getMessage()` could run well, right?

Comment (cmonkey) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: lready add the test code

Comment (tristaZero) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: Sorry, I did not catch you. 
Which condition will make `sqlException.getCause isNull` ?

Comment (cmonkey) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: image

Comment (tristaZero) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: @cmonkey Sorry?
Or what's the difference between `sqlException.getCause().getMessage()` and `cause.getMessage()`?

Comment (cmonkey) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: val exception = new SQLException()
MySQLErrPacketFactory.newInstance(1, exception)
cause.isInstanceOf[SQLException) is true 

cause.asInstanceOF[SQLException) to  sqlException
sqlException.SQLState is null 
sqlException.cuase is not null 
sqlException.getCuase is null

**Comment by kimmking** [comment]
fix ci for code style first.

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java (hunk 1 above)
Comment (tristaZero) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/MySQLErrPacketFactory.java:50]: Hi @cmonkey Thanks for your explanation. Make sense to me.
I will merge this PR after CI passes. Could you give `travis-ci` a look? I found it failed. Thanks a lot.

**Pull request merged**